### ISSUE TITLE
Spurious duplicate identifier error while destructuring

### ISSUE THREAD

**Issue by weswigham** [microsoft/TypeScript#23823]
**Code**

```ts
let a: string;
let b: string;

// Shouldn't be an error, is destructuring not object literal
({ foo: a, foo: b } = { foo: "no" });
```

**Expected behavior:**
No error - I'm just extracting the same name into two different variables.

**Actual behavior:**
"Duplicate identifier" on the second "foo" in the destructuring pattern, also some object literal strict mode error in the same place if the file is in strict mode.

[**Playground Link**](https://www.typescriptlang.org/play/#src=let%20a%3A%20string%3B%0D%0Alet%20b%3A%20string%3B%0D%0A%0D%0A%2F%2F%20Shouldn't%20be%20an%20error%2C%20is%20destructuring%20not%20object%20literal%0D%0A(%7B%20foo%3A%20a%2C%20foo%3A%20b%20%7D%20%3D%20%7B%20foo%3A%20%22no%22%20%7D)%3B%0D%0A)

**Comment by Zzzen** [comment]
Seems fixed in [3.8.3](https://www.typescriptlang.org/play/?ts=3.8.3#src=let%20a%3A%20string%3B%0D%0Alet%20b%3A%20string%3B%0D%0A%0D%0A%2F%2F%20Shouldn't%20be%20an%20error%2C%20is%20destructuring%20not%20object%20literal%0D%0A(%7B%20foo%3A%20a%2C%20foo%3A%20b%20%7D%20%3D%20%7B%20foo%3A%20%22no%22%20%7D)%3B%0D%0A)

**Comment by jakebailey** [comment]
Yes, this was fixed in #35948, making #35939 a dupe of this one.
